dame_flame/post_matching.py

```diff
@@ -0,0 +1,167 @@
+    '''
+    This function returns the CATEs for all specified unit indices
+    
+    Args:
+        return_df : output of DAME or FLAME
+        unit_id : units for which CATE will be computed
+        input_data : matching data
+        treatment_column_name : name of column containing treatment information
+        outcome_column_name : name of column containing outcome information
+    '''
+    #accept int or list
+    if type(unit_ids) is int:
+        unit_ids = [unit_ids]
+    #define relevant output variables
+    MGs = return_df[1]
+    #recover CATEs
+    CATEs = []
+    for i in unit_ids:
+        status = 0
+        for j in MGs:
+            if i in j:
+                df_mmg = input_data.loc[j,[treatment_column_name, 
+                                                outcome_column_name]]
+                status = 1
+                break
+        if status == 0:
+            CATEs.append(np.nan)
+            print('Unit ' + str(i) + " does not have any matches, so can't " \
+                  "find the CATE")
+        else:
+            treated = df_mmg.loc[df_mmg[treatment_column_name] == 1]
+            control = df_mmg.loc[df_mmg[treatment_column_name] == 0]
+            avg_treated = sum(treated[outcome_column_name])/len(treated.index)
+            avg_control = sum(control[outcome_column_name])/len(control.index)
+            CATEs.append(avg_treated - avg_control)
+    #format output
+    if len(CATEs) == 1:
+        CATEs = CATEs[0]
+    return CATEs
+        
+print(CATE(result_new, 0, df))
+
+#%% ATE function
+def ATE(return_df, input_data, treatment_column_name = 'treated',
+         outcome_column_name = 'outcome'):
+    '''
+    This function returns the ATE for the matching data
+    
+    Args:
+        return_df : output of a call to DAME or FLAME
+        input_data : matching data
+        treatment_column_name : name of column containing treatment information
+        outcome_column_name : name of column containing outcome information
+    '''
+    #define relevant output variables
+    MGs = return_df[1]
+    weights = return_df[0]['weights']
+    #recover CATEs
+    CATEs = [0] * len(MGs)
+    for i in range(len(MGs)):
+        group_data = input_data.loc[MGs[i], [treatment_column_name, outcome_column_name]]
+        treated = group_data.loc[group_data[treatment_column_name] == 1]
+        control = group_data.loc[group_data[treatment_column_name] == 0]
+        avg_treated = sum(treated[outcome_column_name]) / len(treated.index)
+        avg_control = sum(control[outcome_column_name]) / len(control.index)
+        CATEs[i] = avg_treated - avg_control
+    #compute ATE
+    weight_sum = 0; weighted_CATE_sum = 0
+    for i in range(len(MGs)):
+        MG_weight = 0;
+        for j in MGs[i]:
+            MG_weight += weights[j]
+        weight_sum += MG_weight
+        weighted_CATE_sum += MG_weight * CATEs[i]
+    return weighted_CATE_sum / weight_sum
+
+print(ATE(result_new, df))
```

Comment:
I think you did use a separate file for this and only ran it separately, but we need to integrate it in the package. I don't think having code outside of a function will error if you leave this here